### PR TITLE
Specify which certificate fields refer to *preferred* certificates.

### PR DESCRIPTION
**Pull request by SanjayVas**
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8)

### CODE REVIEW

**Review by efoxepstein — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8)* status: all files reviewed, 1 unresolved discussion (waiting on @kungfucraig, @SanjayVas, and @zachcwc)

___
*[docs/certificates.md, line 17 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8#-MXm_UVZ9VXxAOsS_I6W:-MXm_UVZ9VXxAOsS_I6X:bg5g3fq) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/42a6c5017aeea5b82555165cbce55bda6a6bc88a/docs/certificates.md#L17)):*
> ```Markdown
> 1.  Within its validity period.
> 1.  Has a later end to its validity period.
> 1.  Has a later start to its validity period.
> ```

Maybe add a note to clarify that while there's an expectation of reasonable time windows (like, no time window fully contains another), this is not enforced at the API level for whatever reason.

"While we expect that certificates with later end dates will also have later start dates, this is not enforced."

**Review by SanjayVas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8)* status: all files reviewed, 1 unresolved discussion (waiting on @efoxepstein, @kungfucraig, and @zachcwc)

___
*[docs/certificates.md, line 17 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8#-MXm_UVZ9VXxAOsS_I6W:-MXnj6v8AiwCs65i_QfC:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/42a6c5017aeea5b82555165cbce55bda6a6bc88a/docs/certificates.md#L17)):*
<details><summary><i>Previously, efoxepstein (Eli Fox-Epstein) wrote…</i></summary><blockquote>

Maybe add a note to clarify that while there's an expectation of reasonable time windows (like, no time window fully contains another), this is not enforced at the API level for whatever reason.

"While we expect that certificates with later end dates will also have later start dates, this is not enforced."
</blockquote></details>

Done.

**Review by efoxepstein — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kungfucraig and @zachcwc)

**Review by kungfucraig — APPROVED** [review]
> Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @zachcwc)

**Review by zachcwc — APPROVED** [review]

File: docs/certificates.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+## Certificate Preference
+
+Some resources have a field that refers to a *preferred* certificate resource.
+The following criteria, in order of most to least significant, determine
+certificate preference within a collection:
+
+1.  Not revoked.
+1.  Within its validity period.
+1.  Has a later end to its validity period.
```
Comment (stevenwarejones) [inline docs/certificates.md:16]: Is there a best-practice reason why we are putting later end certificates with higher precedence above later start certificates? A later-start date implies a more recently created certificate which should have higher precedence.

Comment (SanjayVas) [reply docs/certificates.md:16]: Good question. While it's true that a later start date implies a more recent certificate, what the system is concerned with is reducing the likelihood that a certificate expires prior to the completion of all operations that are using it. Picking the certificate with the longest remaining time to expiration better accomplishes this.

In practice, we expect that certificates with later end dates will also have later start dates, so the precedence between those two items would be moot.

**Pull request merged**